src/CMakeLists.txt

```diff
@@ -32,23 +41,34 @@ execute_process(
 
 message(STATUS "Awkward includes : ${AWKWARD_INCLUDE}")
 message(STATUS "Awkward libraries: ${AWKWARD_LIBRARIES}")
-include_directories(BEFORE "${AWKWARD_INCLUDE}")
 
 find_library(CPU-KERNELS awkward-cpu-kernels REQUIRED HINTS ${AWKWARD_LIBRARIES})
 find_library(LIBAWKWARD awkward REQUIRED HINTS ${AWKWARD_LIBRARIES})
 find_library(LIBDL dl REQUIRED)
-message(STATUS "Libraries: ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL}")
+find_library(LIBARROW libarrow.so.400 REQUIRED HINTS ${PYARROW_LIBRARY_DIRS} NO_DEFAULT_PATH)
+find_library(LIBARROW_PYTHON libarrow_python.so.400 REQUIRED HINTS ${PYARROW_LIBRARY_DIRS})
+message(STATUS "Libraries: ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL} ${LIBARROW} ${LIBARROW_PYTHON}")
+
+
+add_library(pyarrow IMPORTED INTERFACE)
+target_link_libraries(pyarrow INTERFACE ${LIBARROW} ${LIBARROW_PYTHON}) #  ${PYTHON_LIBRARIES})
+target_include_directories(pyarrow INTERFACE "${PYARROW_INCLUDE}")
+target_link_options(pyarrow INTERFACE "-Wl,--unresolved-symbols=ignore-all")
+
+add_library(awkward IMPORTED INTERFACE)
+target_link_libraries(awkward INTERFACE ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL})
+target_include_directories(awkward INTERFACE "${AWKWARD_INCLUDE}")
 
 # minimal awkward+pybind11
 pybind11_add_module(example example/python.cpp example/minimal.cpp)
 set_target_properties(example PROPERTIES CXX_VISIBILITY_PRESET hidden)
-target_link_libraries(example PRIVATE ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL})
+target_link_libraries(example PRIVATE awkward)
 
 # minimal arrow
 add_executable(example_arrow example/arrow.cpp)
-target_link_libraries(example_arrow PRIVATE arrow_shared)
+target_link_libraries(example_arrow PRIVATE pyarrow)
 
 # awkward+pybind11+arrow
 pybind11_add_module(babel babel/python.cpp)
 set_target_properties(babel PROPERTIES CXX_VISIBILITY_PRESET hidden)
-target_link_libraries(babel PRIVATE ${PYARROW_LIBRARIES})
```

Comment:
There was no "PYARROW_LIBRARIES", that was an undefined variable. I think arrow_shared above might have been too.

Reply:
I must have dropped `find(Arrow REQUIRED)` at some point and forgot to clean up the rest... I was trying to figure out if I ever needed the `Arrow` libs from `apt` installs, or if I only ever need the ones shipped in the python package. It is still not clear to me from the docs because the API seems a little different in either case, but unsure.

Reply:
You still have it, but it does not define PYARROW anything.

Reply:
Does pyarrow ship with all of these libs already? This is probably what I'm a bit confused by. Initially, I thought the arrow C++ was only via source or from apt-get installs; and arrow python is only from pip installs. But it seems like both of them have a "base arrow" API that can be used in C++, and I don't necessarily need both. E.G. I could rely only on `pyarrow` and `arrow` from `pip install` for most of what we want to do, I think... I don't know what the separation is.